pkg/v1/tkg/client/management_components.go

```diff
@@ -4,35 +4,54 @@
 package client
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/pkg/errors"
 	"gopkg.in/yaml.v3"
 
 	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/carvelhelpers"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/clusterclient"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/constants"
 	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/log"
 	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/managementcomponents"
 	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/tkgconfigreaderwriter"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/tkgpackagedatamodel"
 	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/utils"
 )
 
-// InstallManagementComponents install management components to the cluster
-func (c *TkgClient) InstallManagementComponents(kubeconfig, kubecontext string) error {
+// InstallOrUpgradeManagementComponents install management components to the cluster
+func (c *TkgClient) InstallOrUpgradeManagementComponents(kubeconfig, kubecontext string, upgrade bool) error {
 	managementPackageRepoImage, err := c.tkgBomClient.GetManagementPackageRepositoryImage()
 	if err != nil {
 		return errors.Wrap(err, "unable to get management package repository image")
 	}
 
+	managementPackageVersion, err := c.tkgBomClient.GetManagementPackagesVersion()
+	if err != nil {
+		return errors.Wrap(err, "unable to get version of management packages")
+	}
+
 	// Override management package repository image if specified as part of below environment variable
 	// NOTE: this override is only for testing purpose and we don't expect this to be used in production scenario
 	mprImage := os.Getenv("MANAGEMENT_PACKAGE_REPO_IMAGE")
 	if mprImage != "" {
 		managementPackageRepoImage = mprImage
 	}
 
+	// Override the version to use for management packages if specified as part of below environment variable
+	// NOTE: this override is only for testing purpose and we don't expect this to be used in production scenario
+	mpVersion := os.Getenv("MANAGEMENT_PACKAGE_VERSION")
```

Comment:
nit: Is it possible to rename to _MANAGEMENT_PACKAGE_VERSION to denote that this is testing only variable?